Configuration/Realm/Realm.xcconfig

```diff
@@ -27,11 +27,11 @@ LD_RUNPATH_SEARCH_PATHS[sdk=iphone*] = $(inherited) @executable_path/Frameworks
 LD_RUNPATH_SEARCH_PATHS[sdk=watch*] = $(inherited) @executable_path/Frameworks @loader_path/Frameworks;
 LD_RUNPATH_SEARCH_PATHS[sdk=appletv*] = $(inherited) @executable_path/Frameworks @loader_path/Frameworks;
 
-OTHER_LDFLAGS[sdk=macosx*] = -lrealm$(REALM_LIBRARY_SUFFIX);
+OTHER_LDFLAGS[sdk=macosx*] = -lrealm$(REALM_LIBRARY_SUFFIX) -framework SystemConfiguration;
```

Comment:
Is there a reason to add these via `OTHER_LDFLAGS` rather than linking to them in the normal way (adding the framework to the Linked Libraries section of the target in the Xcode project)?

One thing that comes to mind is that this means that anyone using the static framework on iOS will have to link to this framework when they add Realm to their project. Perhaps we should be loading this framework at runtime when needed, rather than linking to it statically?